crates/forge/tests/e2e/templates.rs

```diff
@@ -0,0 +1,38 @@
+use std::fs;
+
+use super::common::runner::runner;
+use assert_fs::prelude::PathChild;
+use camino::Utf8PathBuf;
+use forge::Template;
+use packages_validation::check_and_lint;
+use test_case::test_case;
+
+#[test_case(&Template::CairoProgram; "cairo-program")]
+#[test_case(&Template::BalanceContract; "balance-contract")]
+#[test_case(&Template::Erc20Contract; "erc20-contract")]
+fn validate_templates(template: &Template) {
+    let temp_dir = assert_fs::TempDir::new().expect("Unable to create a temporary directory");
+    let package_name = format!("{}_test", template.to_string().replace('-', "_"));
+
+    runner(&temp_dir)
+        .args([
+            "new",
+            "--template",
+            template.to_string().as_str(),
+            &package_name,
+        ])
+        .assert()
+        .success();
+
+    let package_path = temp_dir.child(package_name);
+    let package_path = Utf8PathBuf::from_path_buf(package_path.to_path_buf())
+        .expect("Failed to convert to Utf8PathBuf");
```

Comment:
We can probably hardcode these, see my comment about paths below

Reply:
In this case w we shouldn't hardcode them - note that these the package name is dynamic, depending on specific test case.

Reply:
Test case can have a reference to the template and a path to it. But maybe that's more complicated than the current solution

Reply:
I would stay with current one tbh.